File: sum_of_beauty_of_all_substrings.js
/**
 * @param {string} s
 * @return {number}
 */
const getMinMaxCount = (map) => {
  let min = Infinity,
    max = -Infinity;
  for (let [key, value] of map) {
    min = Math.min(min, value);
    max = Math.max(max, value);
  }
  return [min, max];
};

const beautySum = function (s) {
  let ans = 0;

  for (let i = 0; i < s.length; i++) {
    let hashMap = new Map();
    for (let j = i; j < s.length; j++) {
      if (hashMap.has(s[j])) {
        hashMap.set(s[j], hashMap.get(s[j]) + 1);
      } else {
        hashMap.set(s[j], 1);
      }

      const [min, max] = getMinMaxCount(hashMap);
      ans += max - min;
    }
  }
  return ans;
};

console.log(beautySum("aaabbc")); //Output: 10

// Example 1:

// Input: s = "aabcb"
// Output: 5
/* Explanation: The substrings with non-zero beauty are 
["aab","aabc","aabcb","abcb","bcb"], 
each with beauty equal to 1. */

File: sum_of_beauty_of_all_substrings.py
def get_min_max_count(char_map):
    values = char_map.values()
    return min(values), max(values)

def beautySum(s):
    ans = 0

    for i in range(len(s)):
        char_map = {}
        for j in range(i, len(s)):
            char = s[j]
            char_map[char] = char_map.get(char, 0) + 1
            min_count, max_count = get_min_max_count(char_map)
            ans += (max_count - min_count)

    return ans

# Example usage
print(beautySum("aabcb"))  # Output: 5
